jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ContainerConfiguration.java

```diff
@@ -83,11 +85,18 @@ public Builder setEnvironment(@Nullable Map<String, String> environmentMap) {
       } else {
         Preconditions.checkArgument(!Iterables.any(environmentMap.keySet(), Objects::isNull));
         Preconditions.checkArgument(!Iterables.any(environmentMap.values(), Objects::isNull));
-        this.environmentMap = ImmutableMap.copyOf(environmentMap);
+        this.environmentMap = new HashMap<>(environmentMap);
       }
       return this;
     }
 
+    public void addEnvironment(String name, String value) {
+      if (environmentMap == null) {
+        environmentMap = new HashMap<>();
```

Comment:
(And ditto the others too)

Reply:
We can do it, but it will change the current behavior that we pass null to fields that are never set, so I wasn't sure if it will be safe. I think `entrypoint` should be able to set to null explicitly. `computeEntrypoint()` can return null to indicate inheriting ENTRYPOINT.

Reply:
Ah right - this is necessary then, which means our previous implementation is actually incorrect in that it can never be null and inherit.